agent/agent_endpoint.go

```diff
@@ -890,8 +890,15 @@ func (s *HTTPServer) AgentRegisterService(resp http.ResponseWriter, req *http.Re
 	}
 
 	// Add the service.
-	if err := s.agent.AddService(ns, chkTypes, true, token, ConfigSourceRemote); err != nil {
-		return nil, err
+	idempotent, _ := strconv.ParseBool(req.URL.Query().Get("idempotent"))
```

Comment:
For the sake of consistency with other Consul APIs, you might check for parameter being present as true (similar to what is done in '&passing' or '&stale' for instance) (but handle false as you do still)

/agent/service/register => idempotent=false
/agent/service/register?idempotent => idempotent=true
/agent/service/register?idempotent=true => idempotent=true
/agent/service/register?idempotent=false => idempotent=false